protocols/gossipsub/src/behaviour.rs

```diff
@@ -222,34 +256,45 @@ impl Gossipsub {
     /// Publishes a message with multiple topics to the network.
     pub fn publish_many(
         &mut self,
-        topic: impl IntoIterator<Item = Topic>,
+        topics: impl IntoIterator<Item = Topic>,
         data: impl Into<Vec<u8>>,
     ) {
-        let message = GossipsubMessage {
-            source: self.local_peer_id.clone(),
-            data: data.into(),
-            // To be interoperable with the go-implementation this is treated as a 64-bit
-            // big-endian uint.
-            sequence_number: rand::random(),
-            topics: topic.into_iter().map(|t| self.topic_hash(t)).collect(),
+        let message = match self.build_message(
+            topics.into_iter().map(|t| self.topic_hash(t)).collect(),
+            data.into(),
+        ) {
+            Ok(m) => m,
+            Err(e) => {
+                error!("{}", e);
+                return;
+            }
```

Comment:
Why not have `publish_many` return a `Result<(), PublishManyError>` and then do:

```suggestion
        let message = self.build_message(
            topics.into_iter().map(|t| self.topic_hash(t)).collect(),
            data.into(),
        )?;
```

As this is a library I think it is better to have the caller handle an error.

Reply:
Agreed